### ISSUE TITLE
[CoE Starter Kit - BUG] Admin Sync Template V3 Flows failed in two environments

### ISSUE THREAD

**Issue by G1982C** [microsoft/coe-starter-kit#5104]
### Describe the issue

Since the 22nd March 2023 I have started to see a couple of flow run errors in two different environments. Most of the runs appear to be working without any issue, it is only impacting a couple of runs. It is failing in the 'Get Flows and store them in CoE CDS Entity' Scope at the step 'List Flows as Admin' with the error details 'Object reference not set to an instance of an object'. Full error details with status code 500 below

{
    "statusCode": 500,
    "headers": {
        "Access-Control-Allow-Methods": "GET, PUT, PATCH, DELETE, POST",
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Max-Age": "3600",
        "Access-Control-Expose-Headers": "*",
        "Date": "Sat, 25 Mar 2023 00:37:33 GMT",
        "Content-Length": "275",
        "Content-Type": "application/json"
    },
    "body": {
        "error": {
            "code": 500,
            "message": "Object reference not set to an instance of an object.",
            "source": "asia-001.azure-apim.net",
            "path": "choose[4]\\when[1]",
            "policyId": "",
            "clientRequestId": "cf4d5773-b321-41da-9cd1-c3029e8c5c2a"
        }
    }
}

### Expected Behavior

Flow runs without producing any errors

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.3

### What app or flow are you having the issue with?

Admin - Sync Template v3 (Flows)

### Steps To Reproduce

1. In Prod Governance Environment open Core Components solution
2. Select Cloud Flows
3. Open Admin - Sync Template V3 Flow
4. Find the run failure

**Comment by Jenefer-Monroe** [comment]
What is shown for the input of that action? 
![image](https://user-images.githubusercontent.com/89584709/227956537-8e3bb17c-ae59-4266-8819-5de07dbcf895.png)

**Comment by G1982C** [comment]
Thanks Jenefer. Correct it only seems to be happening on two environments based on the emails I have received notifying me of the errors. 

I have split out my responses based on the two environments - one is the default environment and the other is a sandbox. 

**Default Environment**
<img width="272" alt="CoEDefEnv" src="https://user-images.githubusercontent.com/117096415/227976698-ed0073e0-16ab-4765-82da-10abac5917f9.PNG">
If I look in the PP Admin View the environment does exist
It also exists in the Product page of the PP Admin Center
<img width="309" alt="CoeDefEnv2" src="https://user-images.githubusercontent.com/117096415/227977810-6f95ba44-b44e-4998-8543-2564e708679c.PNG">

**Power Play Box Sandox Environment**
<img width="248" alt="CoeSandBox" src="https://user-images.githubusercontent.com/117096415/227978788-a869e0d5-1df7-4c13-9f3e-765f858fb57a.PNG">
If I look in the PP Admin view the environment exists
It also exists in the Power Platform Admin Center
<img width="439" alt="SandBox2" src="https://user-images.githubusercontent.com/117096415/227979366-60a76cbd-5f02-4655-a52d-bf3fb9793510.PNG">

**Comment by Jenefer-Monroe** [comment]
And if you rerun the flows now, when you are logged in as the user, does it pass?

**Comment by G1982C** [comment]
I have rerun the flow logged into the browser as the service account and it has been running for 2 hours and keep retrying. 

<img width="388" alt="CoEDefEnvError" src="https://user-images.githubusercontent.com/117096415/228044487-782b0029-e3e1-4f4a-be30-5571d911e9f0.PNG">

**Comment by Jenefer-Monroe** [comment]
ok there is definately something wrong then. This action doesnt do anything special beyond normal product behavior. 
Can you please create a new test flow as this user, add this action, and see if it can succeed there? 
If it does, then we'll add on the other settings (Eg pagination) to see if it passes.

**Comment by G1982C** [comment]
Thanks Jenefer. Created a test flow to List Flows as Admin for that environment and it completed in 4 seconds

<img width="367" alt="CoEError3" src="https://user-images.githubusercontent.com/117096415/228048441-7d2c84c2-05ba-441c-ad8a-5bbbb1b0c85f.PNG">

**Comment by Jenefer-Monroe** [comment]
You ran that as the same identity that is runnign the CoE correct? 

Please add these two settings now
1. Pagination of 100000
![image](https://user-images.githubusercontent.com/89584709/228052753-f2610b61-a19b-4106-90c6-fb832353eee6.png)

2. Expand Suspension Info = Yes
![image](https://user-images.githubusercontent.com/89584709/228052490-2ce811c5-8e3b-42f0-9b05-b3668ba3844d.png)

**Comment by G1982C** [comment]
Correct. I am using the same service account that is the owner of the CoE flows for the test flow. 

Just to say that the issue has now resolved itself since creating that test flow and running it. I checked this morning and the last run of the flow had no errors. I then went to that specific failed flow and resubmitted it and it is now running past the 'List Flows as Admin' without any issues. 

Given that this has been an issue for about 5 days (so pretty reliable at failing each day), has the process of creating and running the test flow resolved the issue? Not sure how, but it has kicked something back into action.
